freeipa/src/main/java/com/sequenceiq/freeipa/flow/freeipa/verticalscale/actions/FreeIPAVerticalScaleActions.java

```diff
@@ -0,0 +1,183 @@
+    private StackService stackService;
+
+    @Inject
+    private ResourceService resourceService;
+
+    @Inject
+    private ResourceToCloudResourceConverter cloudResourceConverter;
+
+    @Inject
+    private StackUpdater stackUpdater;
+
+    @Inject
+    private EventBus eventBus;
+
+    @Bean(name = "STACK_VERTICALSCALE_STATE")
+    public Action<?, ?> stackVerticalScale() {
+        return new AbstractFreeIPAVerticalScaleAction<>(FreeIPAVerticalScalingTriggerEvent.class) {
+            @Override
+            protected void doExecute(StackContext ctx, FreeIPAVerticalScalingTriggerEvent payload, Map<Object, Object> variables) {
+                com.sequenceiq.freeipa.api.v1.freeipa.stack.model.scale.FreeIPAVerticalScaleRequest freeIPAVerticalScaleRequest = payload.getRequest();
+                Stack stack = stackService.getByIdWithListsInTransaction(payload.getResourceId());
+                try {
+                    List<Resource> resources = resourceService.findAllByStackId(payload.getResourceId());
+                    List<CloudResource> cloudResources =
+                            resources.stream().map(resource -> cloudResourceConverter.convert(resource)).collect(Collectors.toList());
+                    CloudCredential cloudCredential = credentialConverter.convert(credentialService.getCredentialByEnvCrn(stack.getEnvironmentCrn()));
+                    CloudStack cloudStack = cloudStackConverter.convert(stack);
+                    cloudStack = cloudStackConverter.updateWithVerticalScaleRequest(cloudStack, freeIPAVerticalScaleRequest);
+                    Location location = location(region(stack.getRegion()), availabilityZone(stack.getAvailabilityZone()));
+                    CloudContext cloudContext = CloudContext.Builder.builder()
+                            .withId(stack.getId())
+                            .withName(stack.getName())
+                            .withCrn(stack.getResourceCrn())
+                            .withPlatform(stack.getCloudPlatform())
+                            .withVariant(stack.getPlatformvariant())
+                            .withLocation(location)
+                            .withAccountId(Crn.safeFromString(stack.getResourceCrn()).getAccountId())
+                            .build();
+
+                    FreeIPAVerticalScaleRequest request = new FreeIPAVerticalScaleRequest(
+                            cloudContext,
+                            cloudCredential,
+                            cloudStack,
+                            cloudResources,
+                            freeIPAVerticalScaleRequest);
+
+                    sendEvent(ctx, request);
+                } catch (Exception e) {
+                    LOGGER.error("Failed to Vertical scaling FreeIPA", e);
+                    sendEvent(ctx, STACK_VERTICALSCALE_FINISHED_FAILURE_EVENT.selector(),
+                            new UpscaleFailureEvent(stack.getId(), "Vertical Scale update", Set.of(), Map.of(), e));
```

Comment:
this doesn't look like the right event here. maybe it should be `FreeIPAVerticalScaleFailureEvent`